src/basic/lock-util.c

```diff
@@ -100,11 +109,12 @@ void release_lock_file(LockFile *f) {
                         f->operation = LOCK_EX|LOCK_NB;
 
                 if ((f->operation & ~LOCK_NB) == LOCK_EX)
-                        unlink_noerrno(f->path);
+                        (void) unlinkat_noerrno(f->dir_fd, f->path, 0);
```

Comment:
just make this unlinkat() and dump the unlinkat_noerrno() nonsense, we don't need this anymore